### DISCUSSION TITLE
Request for Comments - LiteLLM Integration

### DISCUSSION THREAD

**Discussion by ishaan-jaff** [ShishirPatil/gorilla#89] [General]
This PR : https://github.com/ShishirPatil/gorilla/pull/84 adds support for Azure, OpenAI, Palm, Anthropic, Cohere, Replicate Models using litellm https://github.com/BerriAI/litellm

**TLDR: gorilla gets:**
- more models out of the box, everything litellm integrates with: https://litellm.readthedocs.io/en/latest/supported/ 
- instant integration of new models by modifying the `model` param (Easy to add models in the future, just change the model param)

**I'm looking to get community feedback on this PR and the liteLLM integration into gorilla** 

Here's a sample of how litellm is used:
```
from litellm import completion

## set ENV variables
# ENV variables can be set in .env file, too. Example in .env.example
os.environ["OPENAI_API_KEY"] = "openai key"
os.environ["COHERE_API_KEY"] = "cohere key"

messages = [{ "content": "Hello, how are you?","role": "user"}]

# openai call
response = completion(model="gpt-3.5-turbo", messages=messages)

# cohere call
response = completion("command-nightly", messages)

# anthropic call
response = completion(model="claude-instant-1", messages=messages)
```

**Comment by ghost** [comment]
Hey @ShishirPatil wanted to bump this and see how we could bring this to the community
